pyvisa_py/gpib.py

```diff
@@ -430,12 +430,12 @@ def gpib_control_ren(self, mode: constants.RENLineOperation) -> StatusCode:
             ):
                 return constants.StatusCode.error_nonsupported_operation
 
-        # INTFC don't have an interface so use the controller
-        ifc = self.interface or self.controller
+        # Commands and remote enable operation are common to the whole bus and
```

Comment:
Should this even be allowed? (writing specifically about the 0x01 command here, but I guess this applies to most of the other stuff this function does) The user might not expect every instrument on their GPIB bus to go into local mode when they call a function on a single instrument. I think it's better to only expose `gpib_control_ren()` for bus controller objects. This way the user is forced to understand that they're doing something that impacts the whole bus. See https://github.com/pyvisa/pyvisa-py/issues/278 for a problem with that today.

I think sending a single instrument to local mode should be done with [`ibloc()`](https://sourceforge.net/p/linux-gpib/code/HEAD/tree/trunk/linux-gpib-user/language/python/gpibinter.c#l409) because that's a device action, not a bus action. I don't see `ibloc()` exposed anywhere in pyvisa-py right now though.